File: app/src/main/java/com/lumiyaviewer/lumiya/render/tex/AutoValue_DrawableTextureParams.java
package com.lumiyaviewer.lumiya.render.tex;

import com.lumiyaviewer.lumiya.slproto.avatar.AvatarTextureFaceIndex;
import java.util.UUID;
import javax.annotation.Nullable;

final class AutoValue_DrawableTextureParams extends DrawableTextureParams {
    private final AvatarTextureFaceIndex avatarFaceIndex;
    private final UUID avatarUUID;
    private final TextureClass textureClass;
    private final UUID uuid;

    AutoValue_DrawableTextureParams(UUID uuid, TextureClass textureClass, @Nullable AvatarTextureFaceIndex avatarTextureFaceIndex, @Nullable UUID avatarUUID) {
        if (uuid == null) {
            throw new NullPointerException("Null uuid");
        }
        this.uuid = uuid;
        if (textureClass == null) {
            throw new NullPointerException("Null textureClass");
        }
        this.textureClass = textureClass;
        this.avatarFaceIndex = avatarTextureFaceIndex;
        this.avatarUUID = avatarUUID;
    }

    @Override
    @Nullable
    public AvatarTextureFaceIndex avatarFaceIndex() {
        return this.avatarFaceIndex;
    }

    @Override
    @Nullable
    public UUID avatarUUID() {
        return this.avatarUUID;
    }

    public boolean equals(Object obj) {
        if (obj == this) {
            return true;
        }
        if (!(obj instanceof DrawableTextureParams)) {
            return false;
        }
        DrawableTextureParams drawableTextureParams = (DrawableTextureParams) obj;
        if (this.uuid.equals(drawableTextureParams.uuid()) && this.textureClass.equals(drawableTextureParams.textureClass()) && (this.avatarFaceIndex != null ? this.avatarFaceIndex.equals(drawableTextureParams.avatarFaceIndex()) : drawableTextureParams.avatarFaceIndex() == null)) {
            return this.avatarUUID == null ? drawableTextureParams.avatarUUID() == null : this.avatarUUID.equals(drawableTextureParams.avatarUUID());
        }
        return false;
    }

    public int hashCode() {
        return (((this.avatarFaceIndex == null ? 0 : this.avatarFaceIndex.hashCode()) ^ ((((this.uuid.hashCode() ^ 1000003) * 1000003) ^ this.textureClass.hashCode()) * 1000003)) * 1000003) ^ (this.avatarUUID != null ? this.avatarUUID.hashCode() : 0);
    }

    @Override
    public TextureClass textureClass() {
        return this.textureClass;
    }

    public String toString() {
        return "DrawableTextureParams{uuid=" + this.uuid + ", textureClass=" + this.textureClass + ", avatarFaceIndex=" + this.avatarFaceIndex + ", avatarUUID=" + this.avatarUUID + "}";
    }

    @Override
    public UUID uuid() {
        return this.uuid;
    }
}

File: app/src/main/java/com/lumiyaviewer/lumiya/render/tex/AutoValue_DrawableTextureParams.kt
package com.lumiyaviewer.lumiya.render.tex

import com.lumiyaviewer.lumiya.slproto.avatar.AvatarTextureFaceIndex
import java.util.UUID

/**
 * Concrete value implementation kept under its historical JVM name so cached
 * data, diagnostics, and any reflective callers retain the same type identity.
 */
internal class AutoValue_DrawableTextureParams(
    private val uuid: UUID,
    private val textureClass: TextureClass,
    private val avatarFaceIndex: AvatarTextureFaceIndex?,
    private val avatarUUID: UUID?,
) : DrawableTextureParams() {
    override fun avatarFaceIndex(): AvatarTextureFaceIndex? = avatarFaceIndex
    override fun avatarUUID(): UUID? = avatarUUID
    override fun textureClass(): TextureClass = textureClass
    override fun uuid(): UUID = uuid

    override fun equals(other: Any?): Boolean =
        other === this || other is DrawableTextureParams &&
            uuid == other.uuid() &&
            textureClass == other.textureClass() &&
            avatarFaceIndex == other.avatarFaceIndex() &&
            avatarUUID == other.avatarUUID()

    // Preserve AutoValue's historical hash algorithm for cache compatibility.
    override fun hashCode(): Int {
        var hash = (uuid.hashCode() xor HASH_SEED) * HASH_SEED
        hash = (hash xor textureClass.hashCode()) * HASH_SEED
        hash = (hash xor (avatarFaceIndex?.hashCode() ?: 0)) * HASH_SEED
        return hash xor (avatarUUID?.hashCode() ?: 0)
    }

    override fun toString(): String =
        "DrawableTextureParams{uuid=$uuid, textureClass=$textureClass, " +
            "avatarFaceIndex=$avatarFaceIndex, avatarUUID=$avatarUUID}"

    private companion object {
        const val HASH_SEED = 1_000_003
    }
}
